generate_bsg_tests.py

```diff
@@ -0,0 +1,208 @@
+            sp = subprocess.run(["diff %s %s > %s" % (ref_v_path, gen_v_path, diffpath)], shell=True)
+
+            if not os.stat(diffpath).st_size == 0:
+                difflist.append(os.path.dirname(gen_v_name))
+            else:
+                passlist.append(os.path.dirname(gen_v_name))
+
+        elif os.path.isfile(ref_v_path):
+            faultlist.append(test_name.removesuffix(".json"))
+
+    list_diffs_and_passes(difflist, passlist, faultlist, test_dir, gen_v_dir)
+
+
+def preprocess_test(filename):
+
+    # Remove comments /* */, (* *) and empty lines
+    sp = subprocess.run(["sed -i -e '/\(\* .* \*\)/d' -e '/^\s*$/d' %s" % filename], shell=True)
+
+
+def list_diffs_and_passes(difflist, passlist, faultlist, test_suite, output_dir):
+
+    summary_name = "%s/%s_summary.md" % (output_dir, os.path.basename(test_suite))
+
+    difflist.sort()
+    passlist.sort()
+    faultlist.sort()
+
+    with open(summary_name, "w") as sum_file:
+        if passlist:
+            sum_file.writelines("### Generated tests that are the same as the reference\n")
+            sum_file.writelines(":heavy_check_mark: " + line + "\n" for line in passlist)
+        if difflist:
+            sum_file.writelines("### Generated tests that differ from the reference\n")
+            sum_file.writelines(":heavy_exclamation_mark: " + line + "\n" for line in difflist)
+        if faultlist:
+            sum_file.writelines("### Tests that failed to be generated\n")
+            sum_file.writelines(":x: " + line + "\n" for line in faultlist)
+        sum_file.close()
+
+    if difflist or faultlist:
+        # Print warning in style of GH actions annotation
+        print("::warning::Some generated tests differ from the reference or were not generated at all. Check the test statuses in the workflow summary or `%s` in the `bsg-tests-diffs` artifacts."
+              % os.path.basename(summary_name))
+
+def main():
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--test-suite-dir", type=Path, default=Path.cwd()/"bsg_micro_designs",
+            help="Test suite directory to generate verilog netlists from.")
+    parser.add_argument("--ref-test-dir", type=Path, default=Path.cwd()/"UHDM-integration-tests"/"tests"/"bsg"/"bsg_micro_designs_results",
+            help="Referential test directory to compare generated verilog netlists with.")
+    parser.add_argument("--output-dir", type=Path, default=Path.cwd()/"build"/"tests"/"bsg_micro_designs",
+            help="Output directory for the generated verilog files.")
+    args = parser.parse_args()
+
+    output_dir = args.output_dir
+    test_suite_dir = args.test_suite_dir
+    ref_test_dir = args.ref_test_dir
+
+    for sub_suite_dir in Path(test_suite_dir).glob("bsg_*"):
+        for json_file in Path(sub_suite_dir).rglob("*.json"):
+            test_name = os.path.basename(os.path.dirname(json_file))
+            gen_tests(test_name, os.path.relpath(sub_suite_dir), ref_test_dir, output_dir)
```

Comment:
Let's stick to `Path`, especially since we already have variable of this type.
```suggestion
            test_name = json_file.parent.name
            gen_tests(test_name, sub_suite_dir, ref_test_dir, output_dir)
```
I've removed the relpath above to support running from different directories than the one expected, but `sub_suite_dir.relative_to(Path.cwd())` should work if you really need a relative path.